services/replay/service.go

```diff
@@ -780,9 +799,22 @@ func (s *Service) handleReplay(w http.ResponseWriter, req *http.Request) {
 	}
 	replay, err := s.replays.Get(id)
 	if err != nil {
-		httpd.HttpError(w, "could not find replay: "+err.Error(), true, http.StatusNotFound)
+		httpd.HttpError(w, "could not find replay: "+id, true, http.StatusNotFound)
+		return
+	}
+
+	finished := req.URL.Query().Get("finished")
+	if finished == "true" && replay.Status != Finished {
+		httpd.HttpError(w, "replay status is not Finished", true, http.StatusBadRequest)
+		return
+	}
```

Comment:
Since previously the `execution-stats` was not a field of the replay why do we need the `finished` query parameter? Seems like if we simply add the  `execution-stats`  field to the replay response struct we remain backwards compatible while providing the new feature we need.

Reply:
Sounds good. The idea was to keep the endpoint response intact and without any changes, but it does make sense to just send the execution stats if they are available.